### DISCUSSION TITLE
CAD Design for GA12YN20-M3 Drive

### DISCUSSION THREAD

**Discussion by Snypes88** [nliaudat/esp32_8ch_motor_shield#48] [Ideas]
Hello everyone, 

I am working on a CAD design to hold the GA12YN20-M3 motors from your other proportional drive project @nliaudat as I am currently looking to implement for floor heating application atm and like running all motors from a single ESP and board (maybe its stupid of me, but I already bought the PCBs from JLPCB and when I added the cost of Homeatic valves I decided there should be a better way :)). 

What is the best way for me to share my design with your project @nliaudat once I have it finished, printed and tested? Also, once made for the 8CH board, I would gladly adapt to the proportional drive if you see potential/like my solution.... 

I'm a huge fan of the work and effort you have put into this project and hope I can make a small contribution 🙏

Best Regards,
Ricardo

**Comment by nliaudat** [comment]
Hi,

You can use the holder you found on proportionnal floor heating valve
project. It works out of the box without the pcb.
For easy soldering the motor pins, i have a simplified pcb that i ll share.


If you want to share your work, you can make a pr


Regards

**Reply by nliaudat** [reply]
![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/043fe1f2-9bad-4cb0-bdf4-9973424bcf35)

**Reply by nliaudat** [reply]
![2023-11-23 15 21 51](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/069f451d-9a76-4c16-b96e-e958380a8ca2)

**Reply by nliaudat** [reply]
I have made many tests with [GA12-N20-M3 15rpm 3V](https://fr.aliexpress.com/item/4000311295036.html?spm=a2g0s.12269583.0.0.18834b0ejbgKSj) and the torque at 3.3v is just enough. In winter, the water heating circuit increase pressure and I had to add a spring to compensate the missing torque. 

That 3.3V motor can in fact be run between 3-6v. If you want to power the valves with esp32_8ch_shield, it could be better to power it with 5V.

**Reply by Snypes88** [reply]
Understood. I already bought all the 3.3V Hi link ACDC regulators (have 5 populated boards from JLC) and will test the 3V plus showing option. I saw the spring in openscad and must say I am a fan of this solution; this should also ensure a certain level of pretension on the M3 nut removing backlash and hopefully making the linear drive run a bit smoother... I opted for the 3D printer bed springs on bed slingers as these are cheap and readily available worse
Worldwide... 

Do you think it is feasible to retrofit the optocouolers and connect via gpio to the esp to circumvent BEMF for positioning, or do you think this is overkill and BEMF is reliable enough? Also, I have 10 valves on one circuit and the documentation states the number of drives can be expanded. Can you elaborate how this could be done? I was thinking maybe the 2 additional motors rely on optical positioning for example?

**Reply by nliaudat** [reply]
Hi, 
To retrofit the opto sensor, it will need a complete rewrite of the pcb cause esp32 has not enought inputs. (Actually, the ADC inputs are paired by 2). It will need an cd74hc4067 analog multiplexer or equivalent. 

For 10 valves, you can use the shift reg out pins.
Look at https://github.com/nliaudat/esp32_8ch_motor_shield/blob/main/shiftregOut.md

**Comment by Snypes88** [comment]
Finally was able to make some progress on the design.... I am considering going with 2 sleds, one that primarily pushes on the valve stem and one that is driven by the motor, with the spring as a "decoupling" body in between...The top body that holds the motor in place and prevents it from moving up will need to be screwed down instead of threaded on, which would be much cooler, but will not work, as then you cannot setup the first sled and motor independently to the entire assembly (if I explain myself well)

**Reply by Snypes88** [reply]
![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/d0e857ee-e995-4c63-b0cd-39eafeadd327)

**Reply by Snypes88** [reply]
![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/fa0cb340-a775-49ee-b47f-07d5c1250abf)

**Reply by Snypes88** [reply]
@nliaudat here is the final design, any thoughts?

![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/0a7a5b2d-abf5-4c77-8c34-eba71fc06630)

![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/7da43275-38c0-424c-a8b4-c5a5a6558456)

**Reply by nliaudat** [reply]
hard to say ... have you 3d design or print file I can test ?

**Reply by Snypes88** [reply]
[ASY_MotorHousing.zip](https://github.com/nliaudat/esp32_8ch_motor_shield/files/13653417/ASY_MotorHousing.zip)

Here is a step file of the model I made... I have changed it from being a bolted design to threaded design 😅

**Reply by nliaudat** [reply]
Hi, 
1st look seems very good ! But I do not understand how you assemble all parts. 
Can you send 3D printer ready separated parts ? I'd like to try it.
I find you design a bit complex for just a motor holder. 
Here you'll find my first proto : 
[diy_valve_actuator_v15.zip](https://github.com/nliaudat/esp32_8ch_motor_shield/files/13653695/diy_valve_actuator_v15.zip)

**Reply by Snypes88** [reply]
The printed parts are only 4 of the bodies, they consist of the threaded bottom with M30, threaded top, the sled with the arms that sits in the bottom M30 threaded body, and the sled in the top that has the hole for an m3 Nut. I have shut down my PC already but can upload individually tomorrow if you like.

**Reply by Snypes88** [reply]
![PXL_20231212_105302997.jpg](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/fa5b1d62-e014-4393-919d-3dcc2cafff82)

![PXL_20231212_105311860.MP.jpg](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/97c1607b-8902-446f-ae14-6667aa045ab0)

![PXL_20231212_131907037.jpg](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/def30849-d2a8-46b2-b670-20ed01a80055)

![PXL_20231212_105318705.jpg](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/d4639f2b-c338-4a5e-ad14-16d7e11ed851)

**Reply by Snypes88** [reply]
Maybe these pictures help give some context... The washer in the bottom sled is not necessary, but the hole ID is smaller than the valve pin diameter and this ensures we have metal to metal contact distributing the force via the washer on a larger surface area then for the plastic... I think the plastic may eventually die if pressed open and shut 100s of times over it's life. 

A "feature" I am also fond off is because the wings on the bottom sled are visible on 4 sides of the M30 threaded body, you can visually inspect the motor operation to make sure it is really pushing the sled all the way down and this actually the valve

**Reply by Snypes88** [reply]
@nliaudat These are the 3D Printed parts... 
[Print_MotorHousing_Threaded_2.zip](https://github.com/nliaudat/esp32_8ch_motor_shield/files/13658463/Print_MotorHousing_Threaded_2.zip)

**Comment by Snypes88** [comment]
> Hi, 1st look seems very good ! But I do not understand how you assemble all parts. Can you send 3D printer ready separated parts ? I'd like to try it. I find you design a bit complex for just a motor holder. Here you'll find my first proto : [diy_valve_actuator_v15.zip](https://github.com/nliaudat/esp32_8ch_motor_shield/files/13653695/diy_valve_actuator_v15.zip)


Seems like a similar concept! Not sure how you actuate the valve as the internals are missing, but if the holes in the side of your body are guides for the linear actuation then yes.

**Comment by Snypes88** [comment]
@nliaudat I have the 3rd iteration now installed and I seem to have corrected the minor pain points I identified from my previous design... now I have a hard endstop at fully open valve by shifting the linear actuators to the top sled rather than the bottom (hence the open window in the upper screw cap). Any thoughts? I was struggling also a bit to define proper BEMF values... originally I had 0.4 set, however, this resulted in the geared motor self destructing and pulling its own drive shaft out of the final gear assembly 😅 I seem to get quite reliable results with 0.25 though with these motors. Does this align with your experience?

**Reply by Snypes88** [reply]
![PXL_20231215_171539596](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/bd32d9bf-aac9-4fd7-850c-722707a7cac6)


here the self destructed motor... Have you ever had this happen?

**Reply by Snypes88** [reply]
And here the updated design, which works without hiccups now for multiple test runs of fully open and fully close....

![PXL_20231215_145201677 MP](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/345a83ff-84d1-47e6-9406-ba020d347c91)

**Reply by Snypes88** [reply]
Lastly in this new design, I have incorporated the winglets on the top sled to align with the Optical distance measuring sensor on your PCB for proportional control... Is there a way to order some of these PCBs, or do you perhaps have a spare lying around that I could buy of you to test the housing design for the all in one version?

![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/42198463/266221be-b854-491d-8482-e9212cd1f00f)

**Reply by nliaudat** [reply]
Sorry but I do not understand why you want to add a endstop button instead of a opto sensor in the board. 

Actually the concept is to have 2 valve actuators solutions with the same enclosure : 

1) Fully controlled valve with opto sensor, BEMF, own mainboard
2) 2 wire motor with endstops controlled by BEMF

**Reply by Snypes88** [reply]
Where have I said that I am adding an end stop button or limit switch? 

By adding the winglets to the top sled I am able to do 2 things. 1 I can implement a hard limit which prevents the motor from pulling the actuator part of the mechanism higher than I would like... It is a physical end stop. And 2, it allows me to prepare for the second design variant which uses the standalone PCB and would measure actual valve position using the Opto coupler. I am aware that these are 2 fully different systems! The screenshot from CAD was just to let you know that I am also looking to help towards designing a housing for the fully proportional project, not only the 8ch board project... Even though I do not have a standalone PCB and have no need for one really as we have floor hearing at home and the centralized controller is the slightly better solution in my specific application scenario.

Currently the micro USB on the adapter housing pictures of the printed design I shared are only there to simplify wiring to power the motor by the 8 ch board. The wires can also go straight to the controller board without USB... By having micro USB for power though, this gives me backwards comparability for the fully proportional, PCB variant as these also use micro USB should I do a retro fit (which will likely not happen as I already have 5 fully populated 8ch pcbs). On your fully proportional GitHub page you were asking for help with the design of the housing in the to do section. Seeing as I am a fan of your work and I am implementing the 8ch board this is where I thought I would start, hence the discussion on this project page... Once I iterated a stable and reliable design I was hoping to port what I have done and adapt the design to the requirements of the fully proportional version with standalone PCB and open a discussion there. However, if you feel this makes no sense or I am wasting the communities time then I am happy to stop sharing updates.

**Reply by nliaudat** [reply]
Do not worry, I very like your contribution !
I have so many question and reviews about the project that I didn't have time to invest in your design.

1) You power the motor with 5v and I mean that's why he brokes. I have done a simple test with 5v, but not long time enough.
2) I ask myself if the ESP-C3-01M is the good choice, cause it cannot be soldered at jlcpcb or for extra 60$ (soldering in the bottom layer of pcb makes extra costs)
3) It is likely that a design with a rectangular PCB in a vertical position could simplify position detection. Several opto or hall effect sensors would be required. But the question is : do we really need an accurate position sensing ?
